Fix LPC GPIOs not using correct bus address

decsny opened the pull request:
Fix the LPC GPIOs having the wrong bus address by putting them all under the same bus node (as they are in hardware all in one GPIO) and renaming them to ports (also the correct term), and unsuppress simple bus reg warnings.

Commit:
dts: nxp,lpc-gpio: Fix binding to group ports

LPC GPIO binding was wrong in that the reg address
on the simple soc bus was given as an index of the gpio ports
within a gpio controller. Fix this by putting the GPIO node
on the simple bus as a single node with the correct base address,
and make the ports children of this node.

Change the driver to get the port number from the reg address
instead of a custom property, and get base address from DT instead
of the SDK macro definition.

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

Commit:
boards: nxp: Remove suppression of LPC DTC warns

DTC warnings are useful and we should keep them enabled.

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

dleach02 commented on dts/bindings/gpio/nxp,lpc-gpio.yaml:
```diff
@@ -1,48 +1,23 @@
-# Copyright (c) 2019-22, NXP
+# Copyright 2024 NXP
```
isn't this supposed to be '2019-22, 2024' ?

decsny replied:
no, this is a new file

decsny replied:
git/github is confused because the new file has the name of the old file, whereas I renamed the old file

danieldegrasse commented on dts/arm/nxp/nxp_rt5xx_common.dtsi:
RT595 devicetree needs to be updated- shouldn't these be a child node of the GPIO parent device? Also, why are we dropping the `resets` property?

danieldegrasse reviewed (commented):
CI failures need to be addressed- but I agree with the idea behind this change, thanks for cleaning this up

danieldegrasse commented on boards/nxp/lpcxpresso55s06/pre_dt_board.cmake:
```diff
@@ -4,8 +4,6 @@
-list(APPEND EXTRA_DTC_FLAGS "-Wno-simple_bus_reg")
 
 # Suppress "unique_unit_address_if_enabled" to handle the following overlaps:
 # - /soc/peripheral@40000000/syscon@0 & /soc/peripheral@40000000/gpio@0
```
```suggestion
# - /soc/peripheral@40000000/syscon@0 & /soc/peripheral@40000000/gpio@4008c000/gpio@0
```

Do we still need this suppression? When I remove this locally I don't see a dtc warning

Commit:
boards: lpcxpresso5s69//ns: Fix mbedtls build err

Fix issue where when building with TFM and PSA client
there is an error from want RSA key type not being
defined in mbedTLS library.

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>
